pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/SystemTopic.java

```diff
@@ -76,4 +76,10 @@ public boolean isCompactionEnabled() {
         // even though is not explicitly set in the policies.
         return !NamespaceService.isHeartbeatNamespace(TopicName.get(topic));
     }
+
+    @Override
+    public boolean isEncryptionRequired() {
+        // System topics are only written by the broker that can't know the encryption context.
```

Comment:
It's tangential, but have we explicitly stated the limitation that system topics are only written to by the broker?

Reply:
AFAIK we haven't state that ONLY the broker (or pulsar internals) write to these topics but they for sure they do so they need this fix